dsc/src/resolve.rs

```diff
@@ -14,14 +13,30 @@ use tracing::{debug, info};
 use crate::util::DSC_CONFIG_ROOT;
 
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
-pub struct Include {
+pub enum IncludeKind {
     /// The path to the file to include.  Path is relative to the file containing the include
     /// and not allowed to reference parent directories.  If a configuration document is used
     /// instead of a file, then the path is relative to the current working directory.
     #[serde(rename = "configurationFile")]
-    pub configuration_file: String,
+    FilePath(String),
+    #[serde(rename = "configurationContent")]
+    Content(String),
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
+pub enum IncludeParametersKind {
     #[serde(rename = "parametersFile")]
-    pub parameters_file: Option<String>,
+    FilePath(String),
+    #[serde(rename = "parametersContent")]
```

Comment:
These `serde renames` complicate debugging.
If a field name is dictated by good user-facing name in json, then it would be good to have it the same in the code.

Reply:
The reason here is to conform to standards in JSON while also conforming to standards in Rust.  In general, it shouldn't be confusing as the only difference is camel casing vs snake casing.

Reply:
Also, because this is an enum, Rust says to use Pascal casing